SpatialBatchNormalization.lua

```diff
@@ -182,3 +182,13 @@ function BN:accGradParameters(input, gradOutput, scale)
       self.gradBias:add(scale, self.buffer2) -- sum over mini-batch
    end
 end
+
+function BN:clearState()
+   self.buffer = nil
```

Comment:
ugh, that's why I suggest to merge this PR in its current state and require every new module/PR to handle clearState. will fix the comments when I have a minute, thanks Francisco.